packages/azure/src/lib/ConsumptionManagement.ts

```diff
@@ -300,34 +302,98 @@ export default class ConsumptionManagementService {
     return usageRowDetails
   }
 
-  private async getConsumptionUsageDetails(
+  private async queryConsumptionUsageDetails(
     startDate: Date,
     endDate: Date,
-    retry = false,
+    includeStart = true,
+    includeEnd = true,
+    maxRetries = 10,
   ): Promise<Array<UsageDetailResult>> {
-    try {
-      const options = {
-        expand: 'properties/meterDetails',
-        filter: `properties/usageStart ge '${startDate.toISOString()}' AND properties/usageEnd le '${endDate.toISOString()}'`,
+    let currentTry = 0
+
+    const errorMsg = `Azure ConsumptionManagementClient UsageDetailRow paging for time range ${startDate.toISOString()} to ${endDate.toISOString()} failed. Reason:`
+
+    while (currentTry <= maxRetries) {
+      currentTry++
+      try {
+        const options = {
+          expand: 'properties/meterDetails',
+          filter: `properties/usageStart ${
+            includeStart ? 'ge' : 'gt'
+          } '${startDate.toISOString()}' AND properties/usageEnd ${
+            includeEnd ? 'le' : 'lt'
+          } '${endDate.toISOString()}'`,
+        }
+        this.consumptionManagementLogger.debug(
+          `Querying consumption usage details from ${startDate.toISOString()} to ${endDate.toISOString()}`,
+        )
+        const usageRows = this.consumptionManagementClient.usageDetails.list(
+          `/subscriptions/${this.consumptionManagementClient.subscriptionId}/`,
+          options,
+        )
+        return await this.pageThroughUsageRows(usageRows)
+      } catch (e) {
+        await this.waitIfRateLimitExceeded(e, errorMsg)
       }
-      const usageRows = this.consumptionManagementClient.usageDetails.list(
-        `/subscriptions/${this.consumptionManagementClient.subscriptionId}/`,
-        options,
+    }
+
+    const err = new Error(
+      `${errorMsg} max retries of ${maxRetries} exceeded, consider setting a smaller chunk size using AZURE_CONSUMPTION_CHUNKS_DAYS`,
+    )
+    this.consumptionManagementLogger.error(err.message, err)
+    throw err
+  }
+
+  private async getConsumptionUsageDetails(
+    startDate: Date,
+    endDate: Date,
+  ): Promise<Array<UsageDetailResult>> {
+    this.consumptionManagementLogger.info(
+      `Getting consumption usage details from ${startDate.toISOString()} to ${endDate.toISOString()}`,
+    )
+    const AZURE = configLoader().AZURE
+
+    // no chunking is wished for
+    if (!AZURE.CONSUMPTION_CHUNKS_DAYS || AZURE.CONSUMPTION_CHUNKS_DAYS === 0) {
```

Comment:
0 is considered a falsy value, so the second condition may not be needed as the first will also return true for a value of 0.
```suggestion
    if (!AZURE.CONSUMPTION_CHUNKS_DAYS) {
```